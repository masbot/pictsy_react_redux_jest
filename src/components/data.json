{
    "images": [
        {
            "id": "1",
            "url": "cat_1.jpeg"
        },
        {
            "id": "2",
            "url": "cat_2.jpeg"
        },
        {
            "id": "3",
            "url": "cat_3.jpeg"
        },
        {
            "id": "4",
            "url": "cat_4.jpeg"
        },
        {
            "id": "5",
            "url": "cat_5.jpeg"
        },
        {
            "id": "6",
            "url": "cat_6.jpeg"
        },
        {
            "id": "7",
            "url": "cat_7.jpeg"
        },
        {
            "id": "8",
            "url": "cat_8.jpeg"
        }
    ]
}