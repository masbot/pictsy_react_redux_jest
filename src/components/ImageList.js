import React, { Component } from 'react';
import ImageItem from './ImageItem';
import data from './data.json';

class ImageList extends Component {

    render() {
        return (
            <div className="image-list">
                {
                    data.images.map(image => {
                        return <ImageItem key={image.id} url={image.url} image={image} onClickImage={this.props.onClickImage}/>
                    })
                }
            </div>
        )
    }
}

export default ImageList;