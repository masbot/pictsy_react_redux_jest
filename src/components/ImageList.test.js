import React from 'react';
import { shallow } from 'enzyme';
import ImageList from './ImageList';
import data from './data.json';

describe('ImageList', () => {

    const imagelist = shallow(<ImageList/>);

    it('renders properly', () => {
        expect(imagelist).toMatchSnapshot();
    });

    describe('when images data is loaded', () => {
        
        it('renders the right number of images', () => {
            expect(imagelist.find('ImageItem').length).toEqual(data.images.length);
        });

        it('creates an ImageItem component', () => {
            expect(imagelist.find('ImageItem').exists()).toBe(true);
        });
    });
});