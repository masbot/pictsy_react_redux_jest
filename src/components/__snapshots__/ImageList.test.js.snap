// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ImageItem
          image={
                    Object {
                              "id": "1",
                              "url": "cat_1.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_1.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "2",
                              "url": "cat_2.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_2.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "3",
                              "url": "cat_3.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_3.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "4",
                              "url": "cat_4.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_4.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "5",
                              "url": "cat_5.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_5.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "6",
                              "url": "cat_6.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_6.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "7",
                              "url": "cat_7.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_7.jpeg"
/>,
        <ImageItem
          image={
                    Object {
                              "id": "8",
                              "url": "cat_8.jpeg",
                            }
          }
          onClickImage={undefined}
          url="cat_8.jpeg"
/>,
      ],
      "className": "image-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "1",
            "url": "cat_1.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_1.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "2",
            "url": "cat_2.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_2.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "3",
            "url": "cat_3.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_3.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "4",
            "url": "cat_4.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_4.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "5",
            "url": "cat_5.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_5.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "6",
            "url": "cat_6.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_6.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "7",
            "url": "cat_7.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_7.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "function",
        "props": Object {
          "image": Object {
            "id": "8",
            "url": "cat_8.jpeg",
          },
          "onClickImage": undefined,
          "url": "cat_8.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ImageItem
            image={
                        Object {
                                    "id": "1",
                                    "url": "cat_1.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_1.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "2",
                                    "url": "cat_2.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_2.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "3",
                                    "url": "cat_3.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_3.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "4",
                                    "url": "cat_4.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_4.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "5",
                                    "url": "cat_5.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_5.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "6",
                                    "url": "cat_6.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_6.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "7",
                                    "url": "cat_7.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_7.jpeg"
/>,
          <ImageItem
            image={
                        Object {
                                    "id": "8",
                                    "url": "cat_8.jpeg",
                                  }
            }
            onClickImage={undefined}
            url="cat_8.jpeg"
/>,
        ],
        "className": "image-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "1",
              "url": "cat_1.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_1.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "2",
              "url": "cat_2.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_2.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "3",
              "url": "cat_3.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_3.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "4",
              "url": "cat_4.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_4.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "5",
              "url": "cat_5.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_5.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "6",
              "url": "cat_6.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_6.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "7",
              "url": "cat_7.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_7.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "image": Object {
              "id": "8",
              "url": "cat_8.jpeg",
            },
            "onClickImage": undefined,
            "url": "cat_8.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
